### DISCUSSION TITLE
Ehr api connection with mirth connect

### DISCUSSION THREAD

**Discussion by kartikey-codes** [nextgenhealthcare/connect#5856] [Q&A]
What should be the configuration of source and destination for connecting with most of the Ehr api , for eg Epic ehr systems

**Comment by kirbykn2** [comment]
What are the requirements for connecting to the API?

**Reply by kartikey-codes** [reply]
OAuth 2.0 authentication 
having specific header files etc...

**Comment by pacmano1** [comment]
Each EHR has their own requirements.  Some API authorizations are backend only, some require interactive approval.  Some require just a token, some require a token followed by credentialing via API call as a named user which simulates a user login.  

Epic documentation is very thorough on what they support. In Epic's case they also use JWTs.  So, like @kirbykn2 alluded to, what does the api documentation for the EHR you are working with say?   

Those requirements define the channel designs. In general though if it is not an authorization type that requires user interaction (an explicit allow to the EHR with a user account in that EHR via a web based approval process you create in Mirth with callbacks), the source might be a dummy javascript reader and the destination(s) http senders.   

A fundamental question his how to maintain an active token, that may be via channel dedicated to refreshing a token on some cadence, it may be a single channel that can get a token but also tests that that token is valid by making some other API call, or a handful of other ways.
